tests/decomposition/test_symbolic_decomposition.py

```diff
@@ -244,12 +207,91 @@ def circuit():
         if not capture_enabled:
             qml.capture.disable()
 
-    def test_non_integer_pow_not_implemented(self):
-        """Tests that NotImplementedError is raised when z isn't a positive integer."""
+    def test_non_integer_pow_not_applicable(self):
+        """Tests that DecompositionNotApplicable is raised when z isn't a positive integer."""
 
         op = qml.pow(qml.H(0), 0.5)
-        with pytest.raises(NotImplementedError, match="Non-integer or negative powers"):
+        with pytest.raises(DecompositionNotApplicable):
```

Comment:
We don't have the same error msg as before any more? It's very tiny pinch but I would prefer having something to tell users who tried and we tested this output as well